src/main/java/com/rev_connect_api/config/SecurityConfig.java

```diff
@@ -30,7 +30,7 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
             .authorizeHttpRequests(authorize -> authorize
                 .requestMatchers("/auth/login", "/users/register", "/h2-console/**", "/attachments/**").permitAll() // allow login and register to everyone
                 .requestMatchers("/users/**", "/posts/**", "/comments/**", "/likes/**", "/profile/?*").hasAnyRole("USER", "ADMIN") // USER or ADMIN roles can access /users/**
-                .anyRequest().authenticated()) // all other requests require authentication
+                .anyRequest().anonymous()) // all other requests require authentication
```

Comment:
https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.AuthorizedUrl.html

Did you mean to change this back? Based on the comment I assumed you wanted to require full authentication for all endpoints not specified. My understanding is that this allows users that do not have an auth token.

Reply:
Any request, not specified is anonymous, so unauthenticated. This returns the correct 403 on attempts to reach endpoints, and the ones specified will be permitted if they have the correct jwt token/roles

Reply:
I don't think this is the case. Sending a get request to http://localhost:8080/api/notAnEndpoint returns a 404 when set to anonymous but returns a 403 when set to authenticated.

Reply:
In the grand scheme of things I think we're fine to merge this for now in either state.

Reply:
that's correct. It should send a 404 since that endpoint doesn't exist. setting to authenticated is forcing authentication on endpoints the server doesn't support.